### ISSUE TITLE
Make ordered as the default behavior

### ISSUE THREAD

**Issue by wookayin** [cgarciae/pypeln#54]
`map()` preserving the order is much more intuitive behavior. Python's builtin Pool executor, ray, joblib, etc. all work in such a way.

I realized that one can still pipe to `pl.process.ordered`, but the documentation is limited and this is quite difficult to use.

```
def slow_identity(x):
   time.sleep(random.random())
   return x

s = list(range(100)) | pl.process.map(slow_identity, workers=N)
list(s)     # should be ordered by default
```

**Comment by cgarciae** [comment]
Hey @wookayin,

Implementing ordering efficiently can get very tricky if you consider multi-stage pipelines containing transformations like `filter` and `flat_map`. The current implementation of `ordered`  is pessimistic and has to wait for all the elements to come in before yielding.

I think the [example](https://cgarciae.github.io/pypeln/api/process/ordered/) from the `ordered` documentation should be able for people to get started, but would be happy to improve if you give some feedback.

**Comment by cgarciae** [comment]
I don't agree that stages should order by default since its a slower operation. We can optimize `ordered` for simple cases and create an `ordered_map` shortcut.

What do you think?

**Comment by gtadamson** [comment]
I agree with @cgarciae that ordering by default would be undesirable if it slowed down the speed of the full pypeln
